charts/latest/azurefile-csi-driver/templates/csi-azurefile-controller.yaml

```diff
@@ -227,6 +242,16 @@ spec:
             {{- end }}
           resources: {{- toYaml .Values.controller.resources.azurefile | nindent 12 }}
       volumes:
+{{- if and .Values.workloadIdentity.clientID .Values.workloadIdentity.tenantID}}
+        - name: azure-identity-token
+          projected:
+            defaultMode: 420
+            sources:
+            - serviceAccountToken:
+                audience: api://AzureADTokenExchange
+                expirationSeconds: 3600
```

Comment:
need to check when token expire, how does this driver do refresh token? is it automatic?

Reply:
According to the [official doc](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview#:~:text=It%27s%20an%20optional%20field%20that%20you%20configure%20to%20prevent%20downtime%0Acaused%20by%20errors%20during%20service%20account%20token%20refresh.%20Kubernetes%20service%20account%20token%20expiry%20isn%27t%20correlated%20with%20Azure%20AD%20tokens.%20Azure%20AD%20tokens%20expire%20in%2024%20hours%20after%20they%27re%20issued.), it will be automatically refreshed. Application Pod don't need to do anything.

Reply:
but we are using old adal lib, will that token be refreshed after 24 hrs?

Reply:
You are right, although the token was refreshed, the client won't refresh its Authorizer header and encountered authentication error, I will take a look at it.

Reply:
adal lib do have the mechanism that will refresh the token after it expired. However there is a bug that will trigger this error under certain circumstances. I've filed a [PR](https://github.com/Azure/go-autorest/pull/719) to fix it, hopefully it can be merged before end-of-life.